### PR TITLE
docs: Add section for filtering by subnet tags in ENI mode

### PR DESCRIPTION
**Pull request by christarazi**
Users frequently ask how to properly filter subnets by tag in ENI mode.
This commit adds a dedicated section to the AWS ENI guide on just that.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

File: Documentation/gettingstarted/aws-eni.rst [hunk 1]
```diff
@@ -68,6 +68,62 @@ Deploy Cilium release via Helm:
...
+two pieces of configuration that must be provided, a custom CNI configuration
+and a field in the Cilium ``ConfigMap``:
+
+Create a CNI configuration
+--------------------------
+
+Create a `cni-config.yaml` file based on the template below. Fill in the
+``subnet-tags`` field:
+
+.. code:: yaml
```
Comment (joestringer) [inline Documentation/gettingstarted/aws-eni.rst]: I think these all need to be `code-block`, `code` is rST while `code-block` is sphinx syntax which allows the sphinx-specific syntax highlighting to work correctly.

```suggestion
.. code-block:: yaml
```

File: Documentation/gettingstarted/aws-eni.rst [hunk 2]
```diff
@@ -68,6 +68,62 @@ Deploy Cilium release via Helm:
...
+       {
+         "cniVersion":"0.3.1",
+         "name":"cilium",
+         "plugins": [
+           {
+             "cniVersion":"0.3.1",
+             "type":"cilium-cni",
+             "eni": {
+               "subnet-tags":{
+                 "foo":"true"
```
Comment (joestringer) [inline Documentation/gettingstarted/aws-eni.rst:103]: I guess there's an implicit assumption here that the user has configured these specific subnet-tags already in some AWS system?

**Review by joestringer — CHANGES_REQUESTED** [review]
One minor nit. I didn't try it out and don't have much AWS background but the example seems like it will be very useful to have around.

File: Documentation/gettingstarted/aws-eni.rst (hunk 2 above)
Comment (christarazi) [reply Documentation/gettingstarted/aws-eni.rst:103]: Yes, that's true. I'll add a note of that above.

**Review by joestringer — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by christarazi** [comment]
Marking ready for merge, as we have approvals and the docs build passed.

**Comment by joestringer** [comment]
Usually we would backport docs fixes like this, but apparently the v1.9 chart doesn't support eni sub-options so we would also need to address that if we were to backport this PR.

**Pull request merged**